pkg/util/log/log.go

```diff
@@ -12,6 +12,12 @@ import (
 	"github.com/weaveworks/common/server"
 )
 
+type contextKey int
+
+const (
+	HeaderMapContextKey contextKey = 0
```

Comment:
I find it a bit weird that this key is declared in `log` package rather than the `api` package. Typically things should be declared close to where they are used 🤔

Reply:
The reason is kind of two fold - the contextKey gets used just as much in the log package as in the api package - and the more important one, putting it in the api package causes an import cycle. I could put the key off somewhere by itself, maybe in the util package, but I wanted to keep it close to where it's used.

Reply:
I see that make sense; didn't thought about the cyclic import.